### DISCUSSION TITLE
Size of the library

### DISCUSSION THREAD

**Discussion by kevinrenskers** [markedjs/marked#2944] [Q&A] [answered]
I'm still on Marked 4.3.0, which is a 37.6 k (11.8k gzipped) dependency.

<img width="476" alt="Screenshot 2023-08-15 at 22 02 45" src="https://github.com/markedjs/marked/assets/229384/f975e5cc-e728-467e-aab2-8293cd08dd4d">

After upgrading to 7.0.3 I noticed that Marked has become significantly bigger, plus I now need to include two extra packages:

<img width="637" alt="Screenshot 2023-08-15 at 22 03 34" src="https://github.com/markedjs/marked/assets/229384/dc5dd62c-7c41-4739-9e38-b9fc0e2b4650">

This now totals 14 k gzipped. Sure, from 11.8 to 14 is only 2.2 k extra, but it's a pretty big percentage. So basically I am wondering what 7.0.3 is doing that 4.3.0 doesn't do? I actually expected the core library to be smaller now that options and functions have been removed, moved to plugins, and support for older Node version has been dropped. How come did it grow in size?

Is it worth upgrading, or should I just stay on 4.3.0 if I care about bundle size?

Thanks for the great library by the way, it's always been flawless for me 👍

**Comment by UziTech** [comment, marked as answer]
The options have been deprecated, not removed yet.

We are using typescript now which seems to be the biggest difference. Most likely it is the source maps that are causing the size difference.

If anyone knows how to reduce the size of those we would appreciate PRs 😁👍

**Reply by kevinrenskers** [reply]
Any idea when the options will actually be removed? That would also help with the size I'd imagine. Now I am basically importing two plugins while their functionality is still built-in as well?

**Reply by UziTech** [reply]
#2936 is a draft PR to remove all deprecated options. It should be ready soon.

**Comment by kevinrenskers** [comment]
So I've waited until Marked 9 was released and tried again.

Now:

![Screenshot 2023-09-29 at 10 30 10](https://github.com/markedjs/marked/assets/229384/c38aba95-e1ba-47c3-b9ff-3baa0b6d5428)

Before (Marked 4.3.0):

<img width="476" alt="Screenshot 2023-08-15 at 22 02 45" src="https://github.com/markedjs/marked/assets/229384/f975e5cc-e728-467e-aab2-8293cd08dd4d">

So it's definitely better than with 7.0.3, but it's still bigger than what it used to be, and what I would expect to be honest. Since so much code has been removed from the core library, shouldn't it be much smaller now?

**Reply by UziTech** [reply]
The problem is mostly moving to typescript. The output JavaScript is larger and we have to include source maps for each file that have the source included.

**Reply by UziTech** [reply]
The other thing making the library so big is that we bundle esm, cjs, umd, and a minified umd version. If we could reduce that to one version the size would be about 1/4 the size

**Reply by kevinrenskers** [reply]
But my app is only importing the actual code, not the source maps, right? And it only imports one version, not four. So that really shouldn't be a reason for the import sizes that I am seeing?

**Reply by UziTech** [reply]
The size you are seeing is the full npm package size. The size your app sends to the user will be about 1/8th of that or less because it will only include the one file you are importing and none of the source maps or other npm files (package.json, etc).

**Reply by kevinrenskers** [reply]
I don't think that's true. I am using this VSCode extension to show the import cost: https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost. As far as I am aware, this is the true size as the user would also download it. 

But I will definitely inspect my build output, see what's going on. Thanks for pointing me into this direction though!

**Reply by UziTech** [reply]
Actually I might be wrong. It looks like that is one file size. Probably mostly typescript adding stuff

**Reply by UziTech** [reply]
The smartypants extension is bigger because it is actually using smarty-pants which follows the spec. The implementation that used to be in marked didn't follow the spec completely. There is a lighter version that is the same as the old marked implementation if you don't care about not following the smartypants spec. https://www.npmjs.com/package/marked-smartypants-lite

**Reply by kevinrenskers** [reply]
Yup I found that one. Its peer dependencies are still set to `"marked": "^4 || ^5"`, but it seems to work. But mostly I was wondering about the core Marked library itself. Like I said: I just expected it to be a lot smaller now that so much has been removed from the core itself.

**Reply by UziTech** [reply]
You can create a PR to update the peerDependencies version it should still work with the latest version of marked.

**Reply by UziTech** [reply]
11.8k (4.3.0) to ~12.4k (current with extensions) seems about right now that smarty pants is actually following spec.
